pkg/bmtpool/bmtpool.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bmtpool
+
+import (
+	"github.com/ethersphere/bee/pkg/swarm"
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+	"github.com/ethersphere/bmt/pool"
+)
+
+var instance pool.Pooler
+
+func init() {
+	instance = pool.New(8, swarm.BmtBranches)
+}
+
+// Get a bmt Hasher instance.
+// Instances are reset before being returned to the caller.
+func Get() *bmtlegacy.Hasher {
+	return instance.Get()
+}
+
+// Put a bmt Hasher back into the pool
+func Put(h *bmtlegacy.Hasher) {
```

Comment:
Side comment (not important): "Release" might be more precise name for 'pool' resource.